src/struct_info.rs

```diff
@@ -42,125 +64,171 @@ impl<'a> StructInfo<'a> {
 
     pub fn builder_creation_impl(&self) -> Result<TokenStream, Error> {
         let init_empties = {
-            let names = self.fields.iter().map(|f| f.name);
+            let names = self.included_fields().map(|f| f.name);
             quote!(#( #names: () ),*)
         };
         let builder_generics = {
-            let names = self.fields.iter().map(|f| f.name);
-            let generic_idents = self.fields.iter().map(|f| &f.generic_ident);
+            let names = self.included_fields().map(|f| f.name);
+            let generic_idents = self.included_fields().map(|f| &f.generic_ident);
             quote!(#( #names: #generic_idents ),*)
         };
-        let StructInfo { ref vis, ref name, ref builder_name, ref core, .. } = *self;
+        let StructInfo {
+            ref vis,
+            ref name,
+            ref builder_name,
+            ref core,
+            ..
+        } = *self;
         let (impl_generics, ty_generics, where_clause) = self.generics.split_for_impl();
         let b_generics = self.modify_generics(|g| {
-            for field in self.fields.iter() {
+            for field in self.included_fields() {
                 g.params.push(field.generic_ty_param());
             }
         });
         let generics_with_empty = modify_types_generics_hack(&ty_generics, |args| {
-            for _ in self.fields.iter() {
+            for _ in self.included_fields() {
                 args.push(syn::GenericArgument::Type(empty_type()));
             }
         });
+        let skip_phantom_generics = self.generics.params.is_empty();
```

Comment:
Not sure if I agree with your reasoning here. This is a generated field that we won't have to touch and the users won't have to see, plus it's private so it won't appear in the docs. I think uniformity is more important than neatness in these cases.

Reply:
This is also the change among them that I’m most dubious about. As it stands, it doesn’t make a great deal of sense, but with what I have in mind in the way of considering the builder a more public thing and worthy of extension, *maybe* it makes sense. Then again, I suppose such extensions can use the existing methods too, so it’s probably not worthwhile after all. Yeah, in the end I’m inclined to agree with you, the increase in code complexity is not warranted.